onnxruntime/core/framework/execution_frame.cc

```diff
@@ -204,21 +226,26 @@ Status IExecutionFrame::GetOutputs(std::vector<OrtValue>& fetches) {
   }
 
   for (size_t idx = 0; idx < num_fetches; ++idx) {
-    fetches[idx] = GetMLValue(fetch_mlvalue_idxs_[idx]);
+    if (release) {
+      GetMLValue(fetch_mlvalue_idxs_[idx], fetches[idx]);
+    } else {
+      fetches[idx] = GetMLValue(fetch_mlvalue_idxs_[idx]);
+    }
   }
 
   return Status::OK();
 }
 
-bool IExecutionFrame::IsOutput(int ort_value_idx) const {
+bool IExecutionFrame::IsOutput(int ort_value_idx) {
```

Comment:
>IsOutput [](start = 22, length = 8)

Seems this can still be const?

Reply:
fetch_mlvalue_idxs_ is not const and is updated between runs. 

---
In reply to: [598071248](https://github.com/microsoft/onnxruntime/pull/7077#discussion_r598071248) [](ancestors = 598071248)